openshift/e2e-add-service-account.py

```diff
@@ -9,6 +9,9 @@
 import sys
 data = list(yaml.safe_load_all(sys.stdin))
 for x in data:
-    if x['kind'] in ('PipelineRun', 'TaskRun'):
-        x['spec']['serviceAccountName'] = sys.argv[1]
+    try:
+        if x['kind'] in ('PipelineRun', 'TaskRun'):
+            x['spec']['serviceAccountName'] = sys.argv[1]
+    except TypeError:
```

Comment:
it's a bit of broad exception catch, what is the error? no spec in x or no kind in x or no sys.argv[1] ?

Reply:
a better fix

```diff
diff --git a/openshift/e2e-add-service-account.py b/openshift/e2e-add-service-account.py
index 6e4c822..c8887c0 100755
--- a/openshift/e2e-add-service-account.py
+++ b/openshift/e2e-add-service-account.py
@@ -9,6 +9,8 @@ import yaml
 import sys
 data = list(yaml.safe_load_all(sys.stdin))
 for x in data:
+    if not x:
+        continue
     if x['kind'] in ('PipelineRun', 'TaskRun'):
         x['spec']['serviceAccountName'] = sys.argv[1]
 print(yaml.dump_all(data, default_flow_style=False))
```

Reply:
Agree!